src/ast/mod.rs

```diff
@@ -793,6 +793,11 @@ pub enum Expr {
     OuterJoin(Box<Expr>),
     /// A reference to the prior level in a CONNECT BY clause.
     Prior(Box<Expr>),
+    /// Scalar variable creation e.g. `[@]foo INT`
+    ScalarVariable {
```

Comment:
hmm, it looks like we're missing implementation in the PR for how the parser actually parses/constructs the ScalarVariable expression or is that intended?